test/unit/runner/suite-runner/decorator-suite-runner.js

```diff
@@ -0,0 +1,129 @@
+const DecoratorSuiteRunner = require('lib/runner/suite-runner/decorator-suite-runner');
+const RunnerEvents = require('lib/constants/runner-events');
+const PrivateEvents = require('lib/runner/private-events');
+const Config = require('lib/config');
+const util = require('../../../util');
+
+describe('runner/suite-runner/decorator-suite-runner', () => {
+    const sandbox = sinon.sandbox.create();
+
+    const makeDecoratorRunner = (suiteRunner, configRootUrl) => {
+        const config = sinon.createStubInstance(Config);
+        const rootUrl = configRootUrl || 'http://localhost/search/touch/';
+
+        config.forBrowser.returns({rootUrl});
+
+        return new DecoratorSuiteRunner(suiteRunner, {}, config);
+    };
+
+    const makeSuiteData = (url) => {
+        url = url || '/testUrl';
+        return {suite: {url}};
+    };
+
+    afterEach(() => sandbox.restore());
+
+    describe('run', () => {
+        it('should call run of decorated suite runner', () => {
+            const suiteRunner = sinon.createStubInstance(SuiteRunner);
+            const decorator = makeDecoratorRunner(suiteRunner);
+
+            return decorator.run()
+                .then(() => assert.calledOnce(suiteRunner.run));
+        });
+    });
+
+    describe('cancel', () => {
+        it('should call cancel of decorated suite runner', () => {
+            const suiteRunner = sinon.createStubInstance(SuiteRunner);
+            const decorator = makeDecoratorRunner(suiteRunner);
+
+            decorator.cancel();
+
+            assert.calledOnce(suiteRunner.cancel);
+        });
+    });
+
+    describe('event decorator', () => {
+        beforeEach(() => sandbox.spy(DecoratorSuiteRunner.prototype, 'emitAndWait'));
+
+        const testSubscribe = (event) => {
+            const suiteRunner = new QEmitter();
+            const decorator = makeDecoratorRunner(suiteRunner, 'http://localhost/search/touch-pad/');
+
+            const eventSpy = sinon.spy().named(event);
+
+            decorator.on(event, eventSpy);
+            suiteRunner.emit(event, makeSuiteData());
+
+            assert.called(eventSpy);
+        };
+
+        const testEmitAndWait = (event) => {
+            const suiteRunner = new QEmitter();
+            const decorator = makeDecoratorRunner(suiteRunner, 'http://localhost/search/touch-pad/');
+
+            suiteRunner.emit(event, makeSuiteData());
+
+            assert.calledWith(decorator.emitAndWait, event);
+        };
+
+        [
+            RunnerEvents.BEGIN_SUITE,
+            RunnerEvents.END_SUITE,
+            RunnerEvents.BEGIN_STATE,
+            RunnerEvents.SKIP_STATE,
+            RunnerEvents.END_STATE,
+            RunnerEvents.WARNING,
+            RunnerEvents.ERROR,
+            PrivateEvents.STATE_RESULT,
+            PrivateEvents.CRITICAL_ERROR
+        ].forEach((event) => {
+            it(`should subscribe suiteRunner to ${event} event`, () => testSubscribe(event));
```

Comment:
should pass all events from original an suite runner to a decorated suite runner